connectors/sources/sharepoint_online.py

```diff
@@ -748,6 +790,72 @@ def _prefix_email(email):
     return _prefix_identity("email", email)
 
 
+def _postfix_group(group):
+    if group is None:
+        return None
+
+    return f"{group} Members"
+
+
+def is_domain_group(user_fields):
+    return user_fields["ContentType"] == "DomainGroup"
+
+
+def is_person(user_fields):
+    return user_fields["ContentType"] == "Person"
+
+
+def _domain_group_id(user_info_name):
+    """Extracts the domain group id.
+
+    The domain group id can have the following formats:
+    - abc|def|domain-group-id
+    - abc|def|some-prefix/domain-group-id
+
+    Returns:
+        str: domain group id
+
+    """
+    if user_info_name is None or len(user_info_name) == 0:
+        return None
+
+    name_parts = user_info_name.split("|")
+
+    if len(name_parts) <= 2:
+        return None
+
+    domain_group_id = name_parts[2]
+
+    if len(domain_group_id) == 0:
+        return None
+
+    if "/" in domain_group_id:
+        domain_group_id = domain_group_id.split("/")[1]
+
+    return domain_group_id
+
+
+async def _emails_and_usernames_of_domain_group(
+    domain_group_id, group_identities_generator, prefix=True
+):
+    """Yield emails and/or usernames for a specific domain group.
+    This function yields both to reduce the number of remote calls made to the group owners or group members API.
+
+    Yields:
+        - Prefixed email or email without prefix
+        - Prefixed user principal name or user principal name without prefix
+
+    """
+    async for identity in group_identities_generator(domain_group_id):
+        email = identity.get("mail")
+        username = identity.get("userPrincipalName")
+
+        if prefix:
```

Comment:
Do we have a use case when we need to prefix `email`/`username`? I think it's better to keep prefix out of the scope of this function.